File: app/src/main/java/com/cheheihome/MainActivity.java
package com.cheheihome;

import android.os.Bundle;
import android.os.StrictMode;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.AbsListView;

import com.cheheihome.supercalendar.R;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

import lecalendar.model.DayModel;
import lecalendar.views.DayView;

/**
 * Created by chanlevel on 15/9/8.
 */
public class MainActivity extends AppCompatActivity implements AbsListView.OnScrollListener {
    LeCalendar gridView;
    DayView dayView;
    boolean flag;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder().penaltyFlashScreen().penaltyLog().build());

        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        gridView = (LeCalendar) findViewById(R.id.gridView);
        dayView = (DayView) findViewById(R.id.dayView);
        dayView.setDaymodel(new DayModel(false, true, "ablgcdp", 0, 0, null, new Date(), false, false, false, false));

        dayView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
//                DayModel dayModel = days.get(310);
//                dayModel.setRoom_num(dayModel.getRoom_num() > 0 ? 0 : 1);
//
//                DayView daycell = (DayView) gridView.getChildAt(10);
//                daycell.setStatus(flag = !flag);
                // dayView.setDayModel(dayModel);

//                gridView.setType(DayView.SElECTTYPE.PRICE);
               // Toast.makeText(MainActivity.this, "", Toast.LENGTH_SHORT).show();
                List<Date> a=gridView.getTotalSelected();
                for (Date d : a)
                    Log.d("super-select", d.toString());

            }
        });


        Calendar b = Calendar.getInstance();
        b.add(Calendar.WEEK_OF_YEAR, -1);
        b.set(Calendar.DAY_OF_WEEK, 1);
        Calendar e = Calendar.getInstance();
        e.add(Calendar.YEAR, 1);
        e.set(Calendar.DAY_OF_WEEK, 7);
        getDates(b, e);
     //   gridView.setDays(days);
        // setGridView();

        //     setGrid();
    }




    void getDates(Calendar begin, Calendar end) {
        Calendar temp = begin;
        // if (this.days == null) this.days = ArrayList<DayModel>();
        while (temp.before(end)) {
            DayModel dayModel = new DayModel(false, false, null, 0, 0, null, temp.getTime(), false, false, false, false);

            dayModel.setRoom_num((int) (Math.random() * 5));
            dayModel.setIsFestival(dayModel.getRoom_num() > 3 ? true : false);
            dayModel.setPrice(dayModel.getRoom_num());
            dayModel.init();
            this.days.add(dayModel);

            temp.add(Calendar.DATE, 1);
        }


    }

    ArrayList<DayModel> days = new ArrayList<>();



    @Override
    public void onScrollStateChanged(AbsListView view, int scrollState) {

    }

    @Override
    public void onScroll(AbsListView view, int firstVisibleItem, int visibleItemCount, int totalItemCount) {
        this.visiblecount = visibleItemCount;
    }

    int visiblecount;

    private static class ViewHolder {
        DayView dayView;
    }
}

File: app/src/main/kotlin/supercalendar/MainActivity.kt
package supercalendar

import android.os.Bundle
import android.os.StrictMode
import android.support.v7.app.AppCompatActivity
import android.widget.AbsListView
import android.widget.Toast
import com.cheheihome.LeCalendar
import kotlinx.android.synthetic.activity_main.*
import com.cheheihome.supercalendar.R

import java.util.ArrayList
import java.util.Calendar
import java.util.Date 

import lecalendar.model.DayModel
import lecalendar.views.DayView

/**
 * Created by chanlevel on 15/9/8.
 */
public class MainActivity : AppCompatActivity(), AbsListView.OnScrollListener {

    var flag: Boolean = false
    var type: DayView.SElECTTYPE = DayView.SElECTTYPE.STATUS

    override fun onCreate(savedInstanceState: Bundle?) {
        StrictMode.setThreadPolicy(StrictMode.ThreadPolicy.Builder().penaltyFlashScreen().penaltyLog().build())

        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)


        val b = Calendar.getInstance()
        b.add(Calendar.DATE, -1)
       // b.set(Calendar.DAY_OF_WEEK, 1)
        val e = Calendar.getInstance()
        e.add(Calendar.YEAR, 1)
        e.set(Calendar.DAY_OF_WEEK, 7)
        getDates(b, e)
        calendar.setDays(days)
        calendar.setSelectListener(object : LeCalendar.LeSelectListener {
            override fun onSelectPosition(status: Boolean, start: Int, end: Int) {

            }

            override fun OnTotalSelected(dates: List<Date>) {
                Toast.makeText(this@MainActivity, dates.size().toString(), Toast.LENGTH_SHORT).show()
            }
        })

    }


    fun getDates(begin: Calendar, end: Calendar) {
        val temp = begin
        // if (this.days == null) this.days = ArrayList<DayModel>();
        while (temp.before(end)) {
            val dayModel = DayModel(false, false, null, 0, 0, null, temp.time, false, false, false, false, false)
            dayModel.room_num = (Math.random() * 5).toInt()
            dayModel.isFestival = if ((Math.random() * 5).toInt()> 3) true else false
            dayModel.price = dayModel.room_num * 10
            dayModel.init()
            this.days.add(dayModel)

            temp.add(Calendar.DATE, 1)
        }


    }

    var days = ArrayList<DayModel>()

    override fun onScrollStateChanged(view: AbsListView, scrollState: Int) {

    }

    override fun onScroll(view: AbsListView, firstVisibleItem: Int, visibleItemCount: Int, totalItemCount: Int) {
        this.visiblecount = visibleItemCount
    }

    var visiblecount: Int = 0


}
